diff --git a/dist/esm/index.js b/dist/esm/index.js
index f35f982..19575e2 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -76,6 +76,9 @@ function transformCode({
   onDirectiveFnsById,
   root
 }) {
+  if (process.platform === "win32" && id.startsWith("/") && !/^\/[A-Za-z]:\//.test(id)) {
+    id = id.slice(1);
+  }
   const url = pathToFileURL(id);
   url.searchParams.delete("v");
   id = fileURLToPath(url).replace(/\\/g, "/");